File: app/src/main/java/com/my/televip/virtuals/SQLite/SQLitePreparedStatement.java
package com.my.televip.virtuals.SQLite;

import com.my.televip.obfuscate.ArgsResolver;
import com.my.televip.obfuscate.Obfuscate;
import com.my.televip.virtuals.tgnet.NativeByteBuffer;

import de.robv.android.xposed.XposedHelpers;

public class SQLitePreparedStatement {

    Object sQLitePreparedStatement;

    public SQLitePreparedStatement(Object sQLitePreparedStatement){
        this.sQLitePreparedStatement = sQLitePreparedStatement;
    }

    public void requery() {
        XposedHelpers.callMethod(sQLitePreparedStatement, Obfuscate.getMethodName("SQLitePreparedStatement","requery"));
    }

    public void step() {
        XposedHelpers.callMethod(sQLitePreparedStatement, Obfuscate.getMethodName("SQLitePreparedStatement","step"));
    }

    public void dispose() {
        XposedHelpers.callMethod(sQLitePreparedStatement, Obfuscate.getMethodName("SQLitePreparedStatement","dispose"));
    }

    public void bindByteBuffer(int index, NativeByteBuffer value) {
        XposedHelpers.callMethod(sQLitePreparedStatement, Obfuscate.getMethodName("SQLitePreparedStatement","bindByteBuffer"), index, value.nativeByteBuffer);
    }

    public void bindLong(int index, long value) {
        XposedHelpers.callMethod(sQLitePreparedStatement, Obfuscate.getMethodName("SQLitePreparedStatement","bindLong"), index, value);
    }

    public void bindInteger(int index, int value) {
        XposedHelpers.callMethod(sQLitePreparedStatement, Obfuscate.getMethodName("SQLitePreparedStatement","bindInteger"), index, value);
    }

}

File: app/src/main/java/com/my/televip/virtuals/SQLite/SQLitePreparedStatement.kt
package com.my.televip.virtuals.SQLite

import com.my.televip.obfuscate.Obfuscate
import com.my.televip.virtuals.tgnet.NativeByteBuffer
import de.robv.android.xposed.XposedHelpers

class SQLitePreparedStatement(@JvmField val sqLitePreparedStatement: Any?) {

    fun requery() {
        val target = sqLitePreparedStatement ?: return
        XposedHelpers.callMethod(target, Obfuscate.getMethodName("SQLitePreparedStatement", "requery"))
    }

    fun bindByteBuffer(index: Int, data: NativeByteBuffer) {
        val target = sqLitePreparedStatement ?: return
        XposedHelpers.callMethod(target, Obfuscate.getMethodName("SQLitePreparedStatement", "bindByteBuffer"), index, data.nativeByteBuffer)
    }

    fun bindLong(index: Int, value: Long) {
        val target = sqLitePreparedStatement ?: return
        XposedHelpers.callMethod(target, Obfuscate.getMethodName("SQLitePreparedStatement", "bindLong"), index, value)
    }

    fun bindInteger(index: Int, value: Int) {
        val target = sqLitePreparedStatement ?: return
        XposedHelpers.callMethod(target, Obfuscate.getMethodName("SQLitePreparedStatement", "bindInteger"), index, value)
    }

    fun step() {
        val target = sqLitePreparedStatement ?: return
        XposedHelpers.callMethod(target, Obfuscate.getMethodName("SQLitePreparedStatement", "step"))
    }
}
